### PR TITLE
grcompiler: new recipe

### PR DESCRIPTION
**Pull request by hoanga**
the following recipe adds support for the [Graphite compiler](http://graphite.sil.org/) 

sample run below:

```
> uname -a
Haiku shredder 1 hrev56098 May 13 2022 08:04:13 x86_64 x86_64 Haiku

> grcompiler --help
Graphite Compiler Version 5.2.1  [release build]
Copyright (c) 2002-2021, by SIL International.  All rights reserved.

usage: grcompiler [options] gdl-file input-font-file [output-font-file] [output-font-name]

Options:
   -c       - compress graphite tables
   -d       - output XML debugger file
   -D       - output all debugger files
   -e       - error message file; gdlerr.txt by default
   -g       - permit and ignore invalid glyph definitions
   -nNNNN   - set name table start location
   -p       - omit pass-avoidance optimizations
   -q       - quiet mode (no messages except on error)
   -vN      - set Silf table version number
   -wNNNN   - ignore warning with the given number
   -wall    - display all warnings
   -offsets - generate xoffset, yoffset, and gpoint glyph attributes
```

### CODE REVIEW

File: app-text/grcompiler/grcompiler-5.2.1.recipe [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ Graphite engine can interpret."
+HOMEPAGE="http://graphite.sil.org/"
+COPYRIGHT="2002 SIL International"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/silnrsi/grcompiler/archive/refs/tags/v$portVersion.tar.gz"
+SOURCE_FILENAME="grcompiler-$portVersion.tar.gz"
+CHECKSUM_SHA256="a125e9d2746638a538e59af8ece2268fda078572f50e82f82d6f95ccb08ed890"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
```
Comment (Begasus) [inline app-text/grcompiler/grcompiler-5.2.1.recipe:14]: Use defaults:
```diff
- ARCHITECTURES="!x86_gcc2 x86 x86_64"
+ ARCHITECTURES="all !x86_gcc2"
```

File: app-text/grcompiler/grcompiler-5.2.1.recipe [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	cmd:cmake
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+	cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$prefix
+	make
```
Comment (Begasus) [inline app-text/grcompiler/grcompiler-5.2.1.recipe:52]: Simplify, use cmakeDirArgs and $commandBinDir
```diff
-	mkdir -p build
-	cd build
-	cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$prefix
-	make
+	cmake -Bbuild -S. -DCMAKE_BUILD_TYPE=Release \
+		$cmakeDirArgs \
+		-DCMAKE_INSTALL_BINDIR=$commandBinDir
+	make -C build $jobArgs
``
	make

File: app-text/grcompiler/grcompiler-5.2.1.recipe [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+INSTALL()
+{
+	cd build
+	mkdir -p $binDir
+	mkdir -p $manDir/man1
+	cp -v compiler/grcompiler $binDir/.
+	cp -v preprocessor/gdlpp $binDir/.
+	cp -v doc/gdlpp.1 $manDir/man1/.
```
Comment (Begasus) [inline app-text/grcompiler/grcompiler-5.2.1.recipe:62]: make install works also:
```diff
-	cd build
-	mkdir -p $binDir
-	mkdir -p $manDir/man1
-	cp -v compiler/grcompiler $binDir/.
-	cp -v preprocessor/gdlpp $binDir/.
-	cp -v doc/gdlpp.1 $manDir/man1/.
+	make -C build install
```

**Review by Begasus — CHANGES_REQUESTED** [review]
There is a testsuite included, so best to add a TEST() section to it

File: app-text/grcompiler/grcompiler-5.2.1.recipe [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libicui18n$secondaryArchSuffix >= 66
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:cmake
+	cmd:pkg_config$secondaryArchSuffix
```
Comment (Begasus) [inline app-text/grcompiler/grcompiler-5.2.1.recipe:44]: Missed one, sort alphabetically :)

File: app-text/grcompiler/grcompiler-5.2.1.recipe [hunk 5]
```diff
@@ -0,0 +1,63 @@
+SUMMARY="SIL Graphite compiler"
+DESCRIPTION="The SIL Graphite compiler builds a Graphite enabled font \
+from a smart font description, written in GDL (Graphite Description Language)\
+and a TrueType font by adding extra TrueType tables to it which the\
+ Graphite engine can interpret."
```
[the remark this answers is not in the document]
Comment (hoanga) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: updated

File: app-text/grcompiler/grcompiler-5.2.1.recipe (hunk 1 above)
Comment (hoanga) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: updated

File: app-text/grcompiler/grcompiler-5.2.1.recipe (hunk 2 above)
Comment (hoanga) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: updated

File: app-text/grcompiler/grcompiler-5.2.1.recipe (hunk 3 above)
Comment (hoanga) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: updated

File: app-text/grcompiler/grcompiler-5.2.1.recipe (hunk 4 above)
Comment (hoanga) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: updated, thanks :)

File: app-text/grcompiler/grcompiler-5.2.1.recipe [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+HOMEPAGE="http://graphite.sil.org/"
+COPYRIGHT="2002 SIL International"
+LICENSE="GNU GPL v2"
```
[the remark this answers is not in the document]
Comment (hoanga) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: it seems that the licensing for the programs can be multi-licensed from what i understand. i have updated the entry to reflect the LGPL v2.1 as i am not sure if there is a way to specify multiple entries in the LICENSE field (doesn't seem so?). i am open to hearing if there is a better suggestion for this.

[the remark this answers is not in the document]
Comment (Begasus) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: See: https://github.com/haikuports/haikuports/wiki/HaikuPorter-BuildRecipes#license
At least the three licenses mentioned should be added: https://github.com/silnrsi/grcompiler/tree/master/license

File: app-text/grcompiler/grcompiler-5.2.1.recipe [hunk 7]
```diff
@@ -0,0 +1,63 @@
+SUMMARY="SIL Graphite compiler"
+DESCRIPTION="The SIL Graphite compiler builds a Graphite enabled font \
+from a smart font description, written in GDL (Graphite Description Language)\
+and a TrueType font by adding extra TrueType tables to it which the \
+Graphite engine can interpret."
+HOMEPAGE="http://graphite.sil.org/"
+COPYRIGHT="2002 SIL International"
```
Comment (Begasus) [inline app-text/grcompiler/grcompiler-5.2.1.recipe:7]: Copyright year: https://github.com/silnrsi/grcompiler/blob/master/license/LICENSING.txt#L3

File: app-text/grcompiler/grcompiler-5.2.1.recipe (hunk 6 above)
[the remark this answers is not in the document]
Comment (hoanga) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: updated, thanks!

File: app-text/grcompiler/grcompiler-5.2.1.recipe (hunk 7 above)
Comment (hoanga) [reply app-text/grcompiler/grcompiler-5.2.1.recipe]: updated

File: app-text/grcompiler/grcompiler-5.2.1.recipe [hunk 8]
```diff
@@ -0,0 +1,66 @@
+SUMMARY="SIL Graphite compiler"
+DESCRIPTION="The SIL Graphite compiler builds a Graphite enabled font \
+from a smart font description, written in GDL (Graphite Description Language)\
+and a TrueType font by adding extra TrueType tables to it which the \
+Graphite engine can interpret."
+HOMEPAGE="http://graphite.sil.org/"
```
Comment (humdingerb) [inline app-text/grcompiler/grcompiler-5.2.1.recipe:6]: That URL doesn't work. Maybe https://scripts.sil.org ?

**Review by Begasus — APPROVED** [review]

**Pull request merged**